utils/error_capture_decorators.py

```diff
@@ -56,19 +56,15 @@ def capture_error_and_close(func: "DecoratorInputFunc[P, T]", error_type: type[B
 
         The raised exception is then suppressed.
         """  # noqa: D401
+
         @functools.wraps(func)
         async def wrapper(self: TeXBotBaseCog, /, *args: P.args, **kwargs: P.kwargs) -> T | None:  # type: ignore[misc] # noqa: E501
-            if not isinstance(self, TeXBotBaseCog):
-                INVALID_METHOD_TYPE_MESSAGE: Final[str] = (  # type: ignore[unreachable]
-                    f"Parameter {self.__name__!r} of any 'capture_error' decorator "
-                    f"must be an instance of {TeXBotBaseCog.__name__!r}/one of its subclasses."
-                )
-                raise TypeError(INVALID_METHOD_TYPE_MESSAGE)
```

Comment:
Ruff's formatter has removed this `isinstance()` check because it saw that `self` was already type-annotated as `TeXBotBaseCog`. The issue is that there is no guarantee that `self` will be the correct type, so it must be checked at runtime.

Perhaps the better solution would be to change the type annotation to `self: object`.

Reply:
Up to you - I don't really mind, is it actually possible to not be an instance of TeXBotBaseCog? either way I don't particularly mind so up to you how you want it done

Reply:
Yeah, it's possible because this is the function for a custom decorator. And so a developer could incorrectly use it on a function that isn't a method within a `TeXBotBaseCog` subclass. I think reinstating the code and changing the type-annotations would be best please.

Reply:
It now complains that object doesn't have an attribute `__name__` - perhaps it is best to just leave it as it was with the ignore unreachable comment?

Reply:
Oh yh, that's correct. We need to accommodate that.

Fix:
```python
if not isinstance(self, TeXBotBaseCog):
    INVALID_METHOD_TYPE_MESSAGE: Final[str] = (
        f"Parameter '{getattr(self, "__name__", None) or "self"}' of any 'capture_error' decorator "
                    f"must be an instance of {TeXBotBaseCog.__name__!r} or one of its subclasses."
                )
                raise TypeError(INVALID_METHOD_TYPE_MESSAGE)